be/src/olap/rowset/segment_creator.cpp

```diff
@@ -59,31 +59,29 @@ Status SegmentFlusher::flush_single_block(const vectorized::Block* block, int32_
     if (block->rows() == 0) {
         return Status::OK();
     }
-    TabletSchemaSPtr flush_schema = nullptr;
     // Expand variant columns
     vectorized::Block flush_block(*block);
     if (_context->write_type != DataWriteType::TYPE_COMPACTION &&
         _context->tablet_schema->num_variant_columns() > 0) {
-        RETURN_IF_ERROR(_expand_variant_to_subcolumns(flush_block, flush_schema));
+        RETURN_IF_ERROR(_expand_variant_to_subcolumns(flush_block));
     }
     bool no_compression = flush_block.bytes() <= config::segment_compression_threshold_kb * 1024;
     if (config::enable_vertical_segment_writer &&
         _context->tablet_schema->cluster_key_idxes().empty()) {
         std::unique_ptr<segment_v2::VerticalSegmentWriter> writer;
-        RETURN_IF_ERROR(_create_segment_writer(writer, segment_id, no_compression, flush_schema));
+        RETURN_IF_ERROR(_create_segment_writer(writer, segment_id, no_compression, _flush_schema));
         RETURN_IF_ERROR(_add_rows(writer, &flush_block, 0, flush_block.rows()));
         RETURN_IF_ERROR(_flush_segment_writer(writer, flush_size));
```

Comment:
you can change signature of this method to avoid adding `_flush_schema` to the class
```cpp
Status _flush_segment_writer(std::unique_ptr<segment_v2::SegmentWriter>& writer,
                                 TabletSchemaSPtr flush_schema = nullptr, int64_t* flush_size = nullptr);
```